### PR TITLE
Rewrite qa_run.pm for kernel regression testing

### PR DESCRIPTION
**Pull request by jtzhao**
Designed a new mechanism to detect whether the test run has finished. Instead of needles, the new qa_run.pm checks if /var/log/qaset/control/DONE exists using shell script.

Besides, I've also fixed the timeout issue of uploading log files by uploading them one by one.

### CODE REVIEW

File: tests/qa_automation/qa_run.pm [hunk 1]
```diff
@@ -31,71 +33,69 @@ sub test_suite() {
...
     type_string "root\n";
     assert_screen "password-prompt", 10;
     type_password;
     type_string "\n";
-    sleep 1;
-
-    # Add Repo - http://dist.nue.suse.com/ibs/QA:/Head/SLE-12-SP1/
+    sleep 2;
+    # Repo & Package
     assert_script_run "zypper --no-gpg-check -n ar -f " . get_var('QA_HEAD_REPO') . " qa_ibs";
```
Comment (cachen) [inline tests/qa_automation/qa_run.pm]: Add step to print all repository is helpful for debugging when pacakge installation failed.

**Comment by jtzhao** [comment]
Wait, I just found a better way to do this. I'll add more commits to it.

**Comment by coolo** [comment]
:+1: happy new year! ;)

**Comment by jtzhao** [comment]
@coolo Thank you~~Nice working with you guys in 2016.

**Comment by sysrich** [comment]
:+1 Happy New Year! from me too, looking forward to the new commits :)

**Comment by jtzhao** [comment]
@sysrich Thank you.
I think the pull request is ready to be merged now.

**Comment by coolo** [comment]
we still need to get the .qcow2 issue fixed though

**Pull request merged**